### PR TITLE
[fmt] update to 11.0.2

### LINKED ISSUE #39660: [fmt] update to <11.0.0>
### Library name

fmt

### New version number

11.0.0

### Other information that may be useful (release notes, etc...)

https://github.com/fmtlib/fmt/releases/tag/11.0.0

### LINKED ISSUE #40000: [fmt] update to 11.0.2
### Library name

fmt

### New version number

11.0.2

### Other information that may be useful (release notes, etc...)

https://github.com/fmtlib/fmt/releases/tag/11.0.2

### PR DESCRIPTION
**Pull request by miyanyan**
Fixes https://github.com/microsoft/vcpkg/issues/40000
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by miyanyan** [comment]
got some error with folly, this's my first time to see such error, will see it tomorrow evening
>CMake Error in CMakeLists.txt:
    Evaluation file to be written multiple times with different content.  This
    is generally caused by the content evaluating the configuration type,
    language, or location of object files:
     D:/github/miyanyan/vcpkg/buildtrees/folly/x64-windows-rel/libfolly.pc

libfolly.pc:
```prefix=D:/github/miyanyan/vcpkg/packages/folly_x64-windows
  exec_prefix=D:/github/miyanyan/vcpkg/packages/folly_x64-windows
  libdir=${exec_prefix}/lib
  includedir=${prefix}/include
  
  Name: libfolly
  Description: Facebook (Folly) C++ library
  Version: 0.58.0-dev
  Cflags: -I${includedir}  /Zc:__cplusplus  -DFMT_SHARED -DBOOST_CONTEXT_NO_LIB= -DBOOST_CONTEXT_DYN_LINK= -DBOOST_CONTEXT_EXPORT=EXPORT -DBOOST_FILESYSTEM_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK=1 -D -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_THREAD_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_USE_DLL -DGLOG_USE_GLOG_EXPORT -DGLOG_USE_GFLAGS -DGLOG_NO_ABBREVIATED_SEVERITIES -DGFLAGS_IS_A_DLL=1 -ID:/github/miyanyan/vcpkg/installed/x64-windows/include
  Libs: -L${libdir} -lfolly
  Libs.private: -lIphlpapi.lib -lWs2_32.lib -lD:/github/miyanyan/vcpkg/installed/x64-windows/lib/zlib.lib -l -lD:/github/miyanyan/vcpkg/installed/x64-windows/lib/libssl.lib -l -lD:/github/miyanyan/vcpkg/installed/x64-windows/lib/libcrypto.lib -l
```

**Comment by jimwang118** [comment]
> got some error with folly, this's my first time to see such error, will see it tomorrow evening
> 
> > CMake Error in CMakeLists.txt:
> > Evaluation file to be written multiple times with different content.  This
> > is generally caused by the content evaluating the configuration type,
> > language, or location of object files:
> > D:/github/miyanyan/vcpkg/buildtrees/folly/x64-windows-rel/libfolly.pc
> 
> libfolly.pc:
> 
> ```
>   exec_prefix=D:/github/miyanyan/vcpkg/packages/folly_x64-windows
>   libdir=${exec_prefix}/lib
>   includedir=${prefix}/include
>   
>   Name: libfolly
>   Description: Facebook (Folly) C++ library
>   Version: 0.58.0-dev
>   Cflags: -I${includedir}  /Zc:__cplusplus  -DFMT_SHARED -DBOOST_CONTEXT_NO_LIB= -DBOOST_CONTEXT_DYN_LINK= -DBOOST_CONTEXT_EXPORT=EXPORT -DBOOST_FILESYSTEM_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK=1 -D -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_THREAD_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_USE_DLL -DGLOG_USE_GLOG_EXPORT -DGLOG_USE_GFLAGS -DGLOG_NO_ABBREVIATED_SEVERITIES -DGFLAGS_IS_A_DLL=1 -ID:/github/miyanyan/vcpkg/installed/x64-windows/include
>   Libs: -L${libdir} -lfolly
>   Libs.private: -lIphlpapi.lib -lWs2_32.lib -lD:/github/miyanyan/vcpkg/installed/x64-windows/lib/zlib.lib -l -lD:/github/miyanyan/vcpkg/installed/x64-windows/lib/libssl.lib -l -lD:/github/miyanyan/vcpkg/installed/x64-windows/lib/libcrypto.lib -l
> ```

This may be caused by multi-threaded compilation. You can add the DISABLE_PARALLEL_CONFIGURE compilation option and then try compiling.

**Comment by dg0yt** [comment]
> This may be caused by multi-threaded compilation. ... `DISABLE_PARALLEL_CONFIGURE`.

This option is for not disabling parallel *configuration* (i.e. CMake). This helps with parallel writes to the source directory (`D:/github/miyanyan/vcpkg/buildtrees/folly/src/...`). 
However, the error points at the build dir (`D:/github/miyanyan/vcpkg/buildtrees/folly/x64-windows-rel/libfolly.pc`).

**Comment by miyanyan** [comment]
> > This may be caused by multi-threaded compilation. ... `DISABLE_PARALLEL_CONFIGURE`.
> 
> This option is for not disabling parallel _configuration_ (i.e. CMake). This helps with parallel writes to the source directory (`D:/github/miyanyan/vcpkg/buildtrees/folly/src/...`). However, the error points at the build dir (`D:/github/miyanyan/vcpkg/buildtrees/folly/x64-windows-rel/libfolly.pc`).

find an issue https://github.com/facebook/folly/issues/2250

**Comment by miyanyan** [comment]
another folly build error is https://github.com/facebook/folly/pull/2255

File: ports/folly/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "folly",
-  "version-string": "2024.07.08.00",
+  "version-string": "2024.07.15.00",
   "port-version": 1,
```
Comment (jimwang118) [inline ports/folly/vcpkg.json:4]:
```suggestion
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by miyanyan** [comment]
> another folly build error is [facebook/folly#2255](https://github.com/facebook/folly/pull/2255)

[https://github.com/fmtlib/fmt/pull/4055](https://github.com/fmtlib/fmt/issues/4036) maybe this will fix

**Comment by miyanyan** [comment]
the arm64_osx check seems failed to start

**Comment by miyanyan** [comment]
https://github.com/facebook/folly/pull/2265

**Comment by jimwang118** [comment]
All features passed with following triplets:

```
x64-windows
x64-windows-static
```
Usage test pass with following triplets:

```
x64-windows
x64-windows-static
```

**Comment by miyanyan** [comment]
> [facebook/folly#2265](https://github.com/facebook/folly/pull/2265)

@jimwang118 my folly patch is based on https://github.com/facebook/folly/pull/2255, but seems https://github.com/facebook/folly/pull/2265 is the chosen one, maybe this pr should not merged now?

**Comment by jimwang118** [comment]
> @jimwang118 my folly patch is based on [facebook/folly#2255](https://github.com/facebook/folly/pull/2255), but seems [facebook/folly#2265](https://github.com/facebook/folly/pull/2265) is the chosen one, maybe this pr should not merged now?

Okay, I have removed the reviewed tag. Maybe you can use vcpkg_download_distfile to directly download this PR as a patch.

**Comment by jimwang118** [comment]
I will change the PR status to Draft. If you have finished editing, please click 'ready for review'.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
> Fixes #40000 

Ooooo nice number.

**Comment by miyanyan** [comment]
hi, if [vcpkg_cmake_configure](https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_cmake_configure) set ```BUILD_SHARED_LIBS``` according to the value of ```VCPKG_LIBRARY_LINKAGE```,  is it still needed to replace ```FMT_SHARED```? I found fmt will set it:
https://github.com/fmtlib/fmt/blob/master/CMakeLists.txt#L349

**Comment by dg0yt** [comment]
If you refer to the `vcpkg_replace_string`: This is needed for projects which don't use CMake.  Mostly msbuild.

**Comment by BillyONeal** [comment]
What dg0yt said. In general we prefer to rely on build system bindings for as little as possible as not every downstream build system listens to such bindings. That's why this replacement was in place before.

**Pull request merged**